src/Command/BackupDbCommand.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace App\Command;
+
+use Doctrine\DBAL\Tools\DsnParser;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Process\ExecutableFinder;
+use Symfony\Component\Process\Process;
+
+#[AsCommand(
+    name: 'app:backup-db',
+    description: 'Add a short description for your command',
+)]
+class BackupDbCommand extends Command
+{
+    public function __construct(
+        private readonly Filesystem $filesystem,
+        private readonly EntityManagerInterface $entityManager,
+        #[Autowire(param: 'kernel.project_dir')]
+        private readonly string $projectDir,
+        #[Autowire(param: 'DATABASE_URL')]
+        private readonly string $dsn,
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        $mysqldump = (new ExecutableFinder())->find('mysqldump');
```

Comment:
I would inject a parameter with a null value, so you can override it (for example if the mysql-dump binary is not within this container and you need to call something outside php

> Ex: docker compose run --rm mysql mysql_dump".